### PR TITLE
Web app manifest for adding to Android home screen

### PR DESCRIPTION
**Pull request by jancborchardt**
Downstreaming a contribution thanks to @Rudloff! :tada: :)  His original comment:

> I added a web app manifest.
> It allows users to add the app to the Android homescreen (see https://developer.chrome.com/multidevice/android/installtohomescreen).

@LukasReschke can you check out the unit tests?

Please review @nextcloud/android :robot:  :)

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
Seems okay to me, while looking at the reference this is a legacy way and the following should be used: https://developers.google.com/web/fundamentals/engage-and-retain/app-install-banners/ which would also allow pointing to the native app, but also comes with some restrictions (site served via HTTPS is a must)

**Comment by jancborchardt** [comment]
@AndyScherzinger If it has additional restrictions then maybe let’s ideally do it in a separate pull request. Then we for sure have the base functionality in and can improve based on it. :)

**Comment by AndyScherzinger** [comment]
@jancborchardt yes, perfectly fine with me :) Just wanted to point out that Google already released a "new" way of doing this.

**Comment by jancborchardt** [comment]
@AndyScherzinger yup, cool! :) So can you approve this? ;)

**Comment by AndyScherzinger** [comment]
@jancborchardt oh yeah, absolutely :)

**Review by AndyScherzinger — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
Nice! Tested and works 👍

**Comment by jancborchardt** [comment]
@AndyScherzinger could you open a new issue with the new way to do it? Then we can look into if it’s something we also want to do. :)

**Comment by jancborchardt** [comment]
@AndyScherzinger whenever you have time and if you still think it’s relevant, it would be cool if you can open an issue about the new way to do it. :)

File: lib/public/AppFramework/Http/EmptyContentSecurityPolicy.php [hunk 1]
```diff
@@ -336,6 +336,7 @@ public function disallowChildSrcDomain($domain) {
 	public function buildPolicy() {
 		$policy = "default-src 'none';";
 		$policy .= "base-uri 'none';";
+		$policy .= "manifest-src 'self';";
```
Comment (Treora) [inline lib/public/AppFramework/Http/EmptyContentSecurityPolicy.php]: Should this really have been added to **Empty**ContentSecurityPolicy? There is also `ContentSecurityPolicy.php`, which has the *default* policy, so (if this line is required) I would expect it to be added there instead.

Due to the current choice, when an app runs `$response->setContentSecurityPolicy(new EmptyContentSecurityPolicy())`, it still ends up allowing manifest links. Perhaps not obviously problematic, but at least contrary to reasonable expectation, and seemingly unnecessary.

Comment (jancborchardt) [reply lib/public/AppFramework/Http/EmptyContentSecurityPolicy.php]: Hmmm @rullzer @MorrisJobke can you help clear this up?

**Pull request merged**